python/lsst/skymap/tractBuilder.py

```diff
@@ -0,0 +1,461 @@
+            PatchInfo of patch whose inner bbox contains the specified coord
+
+        Raises
+        ------
+        LookupError
+            If coord is not in tract or we cannot determine the
+            pixel coordinate (which likely means the coord is off the tract).
+        """
+        try:
+            pixel = wcs.skyToPixel(coord)
+        except (lsst.pex.exceptions.DomainError, lsst.pex.exceptions.RuntimeError):
+            # Point must be way off the tract
+            raise LookupError("Unable to determine pixel position for coordinate %s" % (coord,))
+        pixelInd = geom.Point2I(pixel)
+        if not self._tractBBox.contains(pixelInd):
+            raise LookupError("coord %s is not in tract %s" % (coord, tractId))
+        # This should probably be the same as above because we only
+        # care about the INNER dimensions.
+        patchInd = tuple(int(pixelInd[i]/self._patchInnerDimensions[i]) for i in range(2))
+        return self.getPatchInfo(patchInd, wcs)
+
+    def findPatchList(self, wcs, coordList):
+        """Find patches containing the specified list of coords.
+
+        Parameters
+        ----------
+        wcs : `lsst.afw.image.SkyWcs`
+            The WCS of the associated tract.
+        coordList : `list` of `lsst.geom.SpherePoint`
+            ICRS sky coordinates to search for.
+
+        Returns
+        -------
+        result : `list` of `lsst.skymap.PatchInfo`
+            List of PatchInfo for patches that contain, or may contain, the
+            specified region. The list will be empty if there is no overlap.
+
+        Notes
+        -----
+        **Warning:**
+
+        - This may give incorrect answers on regions that are larger than a
+          tract.
+
+        - This uses a naive algorithm that may find some patches that do not
+          overlap the region (especially if the region is not a rectangle
+          aligned along patch x,y).
+        """
+        box2D = geom.Box2D()
+        for coord in coordList:
+            try:
+                pixelPos = wcs.skyToPixel(coord)
+            except (lsst.pex.exceptions.DomainError, lsst.pex.exceptions.RuntimeError):
+                # the point is so far off the tract that its pixel position cannot be computed
+                continue
+            box2D.include(pixelPos)
+        bbox = geom.Box2I(box2D)
+        bbox.grow(self.getPatchBorder())
+        bbox.clip(self._tractBBox)
+        if bbox.isEmpty():
+            return ()
+
+        llPatchInd = tuple(int(bbox.getMin()[i]/self._patchInnerDimensions[i]) for i in range(2))
+        urPatchInd = tuple(int(bbox.getMax()[i]/self._patchInnerDimensions[i]) for i in range(2))
+        return tuple(self.getPatchInfo((xInd, yInd), wcs)
+                     for xInd in range(llPatchInd[0], urPatchInd[0]+1)
+                     for yInd in range(llPatchInd[1], urPatchInd[1]+1))
+
+    def getPatchBorder(self):
+        return self._patchBorder
```

Comment:
At the moment, yes.  Both legacy and cells have a border region, it's just one is an arbitrary number of pixels, the other is an arbitrary number of cells.

Reply:
Hmm.  If it means different things for different builders, I would be inclined to get rid of it if we possibly can.

Reply:
It is always pixels that are returned here.  It's just how it's specified in the config.  You can specify the number of pixels directly (legacy) or the number of cells (cells), and since each cell has an explicit size you know how big the border is.

Reply:
Ok, that's not too bad.  Still, if nothing calls it, I'd vote for dropping it.

Reply:
It's used in how to define the patches that make the tract.  Dropping it would require some other big changes.

Reply:
Ok, I'd hoped because the patches have different geometry on the edges, it might not have been needed, but if it is, fine.